plugin/pkg/shoot/validator/admission.go

```diff
@@ -233,13 +233,15 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, _ adm
 	// On DELETE operations we want to verify that the Shoot is not in Restore or Migrate phase
 
 	// Exit early if shoot spec hasn't changed
-	if a.GetOperation() == admission.Update {
-		old, ok := a.GetOldObject().(*core.Shoot)
+	if a.GetOperation() == admission.Update || a.GetOperation() == admission.Delete {
```

Comment:
though now the comment above this block seems to be no longer valid, as the respective logic has moved to the subsequent block.